app/Admin/src/Http/Controllers/Auth/AuthController.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace App\Admin\Http\Controllers\Auth;
+
+use App\Admin\Http\Requests\Auth\LoginRequest;
+use App\Admin\Http\Requests\Auth\LogoutRequest;
+use App\Platform\Enums\RoleName;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Validation\ValidationException;
+use Knuckles\Scribe\Attributes\Authenticated;
+use Knuckles\Scribe\Attributes\BodyParam;
+use Knuckles\Scribe\Attributes\Endpoint;
+
+class AuthController
+{
+    /**
+     * @throws ValidationException
+     */
+    #[Endpoint('Admin/Login', <<<'DESC'
+    This endpoint enable users with admin role to sign in
+ DESC)]
+    #[BodyParam('email', 'password', required: true)]
+    public function login(LoginRequest $request)
+    {
+        $credentials = $request->only('email', 'password');
+
+        if (Auth::attempt($credentials)) {
+
+            $user = Auth::user();
+
+            if ($user->hasRole(RoleName::ADMIN)) {
+                $user = $request->user();
+                $token = $user->tokens->where('name', $user->email)->first();
+
+                if ($token) {
+                    $token->delete();
+                }
```

Comment:
You could probably go for a delete directly instead of looking for the record and then deleting it. You are avoiding 1 extra query instead of triggering 2 queries.